pennylane/transforms/resolve_dynamic_wires.py

```diff
@@ -0,0 +1,198 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains a transform for resolving dynamic wires into real wires.
+"""
+from collections.abc import Hashable, Sequence
+from typing import Optional
+
+from pennylane.allocation import Allocate, Deallocate
+from pennylane.measurements import measure
+from pennylane.tape import QuantumScript, QuantumScriptBatch
+from pennylane.typing import PostprocessingFn, Result, ResultBatch
+
+from .core import transform
+
+
+class _WireManager:
+    """Handles converting dynamic wires into concrete values."""
+
+    def __init__(self, zeroed=(), any_state=(), min_int=None):
+        self._zeroed = list(zeroed)
+        self._any_state = list(any_state)
+        self._loaned = {}  # wire to final register type
+        self.min_int = min_int
+
+    def get_wire(self, require_zeros, restored):
+        """Retrieve a concrete wire label from available registers."""
+        if not self._zeroed and not self._any_state:
+            if self.min_int is None:
+                raise ValueError("no wires left to allocate.")
+            self._zeroed.append(self.min_int)
+            self.min_int += 1
+        if require_zeros:
+            if self._zeroed:
+                w = self._zeroed.pop()
+                self._loaned[w] = "zeroed" if restored else "any_state"
+                return w, []
+            w = self._any_state.pop()
+            self._loaned[w] = "zeroed" if restored else "any_state"
+            m = measure(w, reset=True)
```

Comment:
It seems that, when `min_int` is present instead of a explicit sequence of wires, when zeroed wires are requested, we'd just be measure-resetting the same wire over and over again instead of incrementing `min_int` and getting a new wire?

Reply:
We increment `min_int` when everything has been loaned out.

Reply:
Let's say the device has many extra clean wires available. You request one wires, do something on it, returns it without cleaning it up, so it goes back into the "any_state" pool, and then you request another clean wire. Now it seems that the default behaviour is to take that used wire from the "any_state" pool and reset it, as opposed to just getting another clean wire from the device, which I feel would make more sense in most cases.

Reply:
I just feel that this way, there is some inconsistency in the behaviour based on how this is initialized. Let's say the device has 10 wires in total, and the circuit uses 5 wires. If the user says `zeroed=[5,6,7,8,9]`, then we keep taking from the pool of clean wires, but if the user says `min_int=5`, which to the user means exactly the same thing, we start resetting and reusing the same wire over and over again.

Reply:
So the difference between `zeroed = [5,6,7,8,9]` and `min_int=5` will only occur when something like `allocate(6)` happens. `zeroed=[5,6,7,8,9]` would error out, but `min_int=5` would loan out `[5,6,7,8,9,10]`.

Reply:
I think this makes sense as is: if a user knows that there are wires that should be used, they can provide them (in the integers-only case they can do `range(min_int, max_int`), and otherwise they can provide `min_int` to indicate that some extra wires may be used, but frugally.
So I'd argue that the current behavior allows users to choose between two modes already, rather than `min_int` being an alias for the `zeroed` input /a way to not give a fixed number of wires in `zeroed`. Does that make sense?